packages/lib/tagOnResponse/cache.ts

```diff
@@ -0,0 +1,27 @@
+import { revalidateTag } from "next/cache";
+
+interface RevalidateProps {
+  id?: string;
+  responseId?: string;
+  functionName?: string;
+}
+
+export const tagOnResponseCache = {
+  tag: {
+    byResponseIdAndId(responseId: string, id: string) {
+      return `responses-${responseId}-tagOnResponses-${id}`;
+    },
+    byFunctionName(functionName: string) {
```

Comment:
Can you explain the reason for the `functionName`; why do we need this?

Reply:
Done, replaced with byEnvironmentId